File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/BetAmountStrategy.java
package com.panopset.blackjackEngine;

public class BetAmountStrategy {

  private final BlackjackGameEngine bge;
  private final BlackjackConfiguration config;

  public BetAmountStrategy(BlackjackGameEngine bge) {
    this.bge = bge;
    this.config = bge.getConfig();
  }

  public int adjust() {
    int betAmount = bge.getNextBet();
    if (bge.isAutomatic()) {
      if (config.isCountVeryPositive()) {
        betAmount = config.getLargeBetInWholeDollars() * 100;
      }
      if (config.isBetIdeaDoubleAfterBust() && bge.isBustedPriorHand()) {
        betAmount = betAmount * 2;
      }
      if (config.isBetIdeaLetItRideAfterTwoWins() && bge.getStreak() > 1) {
        betAmount = betAmount * 2;
      }
    }
    return betAmount;
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/BetAmountStrategy.kt
package com.panopset.blackjackEngine

class BetAmountStrategy(private val bge: BlackjackGameEngine) {
    private val config: BlackjackConfiguration = bge.config

    fun adjust(): Int {
        var betAmount = bge.getNextBet()
        if (bge.isAutomaticRunning()) {
            if (bge.isCountVeryPositive()) {
                betAmount = config.getLargeBetInWholeDollars() * 100
            }
        }
        return betAmount
    }
}
